src/main/groovy/wooga/gradle/unity/utils/internal/UnityHub.groovy

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2018 Wooga GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wooga.gradle.unity.utils.internal
+
+import groovy.json.JsonSlurper
+
+class UnityHub {
+
+    static HUB_INSTALL_LOCATION = "secondaryInstallPath.json"
+    static HUB_MANUAL_EDITORS = "editors.json"
+    static HUB_DEFAULT_EDITOR = "defaultEditor.json"
+
+    static File DEFAULT_UNITY_HUB_INSTALL_LOCATION_MAC_OS() {
+        new File("/Applications/Unity/Hub/Editor")
+    }
+
+    static File DEFAULT_UNITY_HUB_INSTALL_LOCATION_WINDOWS() {
+        new File("${System.env.get('ProgramFiles')}\\Unity\\Hub\\Editor")
+    }
+
+    private static File UNITY_HUB_CONFIG_PATH_MAC_OS() {
+        new File("${System.env.get('HOME')}/Library/Application Support/UnityHub")
+    }
+
+    private static File UNITY_HUB_CONFIG_PATH_WINDOWS() {
+        new File("${System.env.get('HOMEPATH')}\\AppData\\Roaming\\UnityHub")
+    }
+
+    static File getConfigLocation() {
+        File unityHubConfigPath = null
+        String osName = System.getProperty("os.name").toLowerCase()
```

Comment:
You check for `os.name` a lot, where is this property set? Is it something provided by Gradle or is it part of machines setup you do?

Reply:
That is a Java system variable. This is set by the runtime.